dv-export/python/query-changes/main.py

```diff
@@ -0,0 +1,149 @@
+                        help="Data Version number the script should export.")
+    parser.add_argument('-e', '--export_uuid', type=str,
+                        help="Optional export UUID to retrieve export metadata for. "
+                             "If provided, a DV export job will not be scheduled.",
+                        required=False)
+
+    parsed_args = parser.parse_args()
+    logger.info("Arguments parsed. %s", ", ".join(f"{arg}: {value}" for arg, value in vars(parsed_args).items()))
+
+    if parsed_args.list_dv:
+        logger.info("Utility was called with list_dv flag. No other arguments are required.")
+        return parsed_args
+
+    if not parsed_args.query_path:
+        raise Exception("Query path is required.")
+
+    if parsed_args.export_uuid:
+        logger.info("Export UUID provided. Will not schedule a DV export job.")
+        return parsed_args
+
+    if not parsed_args.data_version:
+        raise Exception("At least one of data_version or export_uuid must be provided.")
+
+    return parsed_args
+
+
+def load_config() -> dict[str, Any]:
+    config = {
+        **dotenv_values('.env.query-changes'),
+        **dotenv_values('.env.visier-auth'),
+        **dotenv_values('.env')
+    }
+    logger.info("Configuration was loaded.")
+    return config
+
+
+def get_filter_property(query):
+    filters = query.get(FILTERS, [{}])
+    member_set = filters[0].get(MEMBER_SET, {})
+    values = member_set.get(VALUES, {})
+    included = values.get(INCLUDED)
+
+    if included is None or not included.startswith('${{') or not included.endswith('}}'):
+        logger.warning("Filter property not found in query. History will be fetched without filters.")
+        return None
+    filter_property = included[3:-2]
+    logger.info(f"Filter property is {filter_property}")
+    return filter_property
+
+
+def main() -> None:
+    args = parse_args()
+    config = load_config()
+
+    auth = make_auth(env_values=OrderedDict(config))
+    with VisierSession(auth) as session:
+        dv_manager = DVManager(session,
+                               bool(config[DV_SAVE_EXPORT_FILES_ON_DISK]),
+                               config[DV_EXPORT_FILES_PATH],
+                               job_status_poll_interval_sec=int(config[DV_JOB_STATUS_POLL_INTERVAL_SECONDS]),
+                               job_timeout_sec=int(config[DV_JOB_TIMEOUT_SECONDS]))
+
+        # List available data versions for export
+        if args.list_dv:
+            data_versions = dv_manager.get_data_versions()
+            logger.info(f"Data versions fetched successfully:\n {json.dumps(data_versions, indent=2)}")
+            return
+
+        # Execute export job if export UUID is not provided
+        if args.export_uuid is None:
+            export_uuid = dv_manager.execute_export_job(args.base_data_version, args.data_version)
+        else:
+            export_uuid = args.export_uuid
+
+        # Load query from file
```

Comment:
Consider looping over all query files if the path provided is a directory

Reply:
Added looping. To simplify main.py moved all logic to CommandHandler class.